cicecore/cicedynB/general/ice_forcing.F90

```diff
@@ -967,9 +966,6 @@ subroutine read_data_nc (flag, recd, yr, ixm, ixx, ixp, &
 
       call ice_timer_stop(timer_readwrite)  ! reading/writing
 
-#else
-      field_data = c0 ! to satisfy intent(out) attribute
-#endif
```

Comment:
In principle it should not reach this part, however if it for whatever reason reaches this part anyway then it should fail in the call to ice_open_nc as the netcdf file cannot be read. Here should be an abort and a cpp flag.

read_data_nc do not use the netcdf library therefore it should be able to run through without warnings.

Reply:
If read_data_nc doesn't use the netcdf library at all then it shouldn't have the _nc suffix on the subroutine name.  I think this routine has evolved over time.

The field_data=c0 line was necessary years ago because a compiler (I don't remember which one, most likely intel) wouldn't make it through the build without it, when the cpp was not invoked.

Reply:
We do not need the protection of the USE_NETCDF or the #else here.  The code will build fine, there are no calls directly to netcdf in this subroutine.  What it does is call _nc routines which will abort if USE_NETCDF is not set.  So anytime this routine is called when USE_NETCDF is set, it should work.  Anytime this routine is called when USE_NETCDF is not set, it will abort in a lower level subroutine.  I think that's the right behavior.